src/libraries/System.Diagnostics.FileVersionInfo/src/System/Diagnostics/FileVersionInfo.Windows.cs

```diff
@@ -12,62 +12,36 @@ private unsafe FileVersionInfo(string fileName)
         {
             _fileName = fileName;
 
-            uint handle;  // This variable is not used, but we need an out variable.
-            uint infoSize = Interop.Version.GetFileVersionInfoSizeEx(
-                (uint)Interop.Version.FileVersionInfoType.FILE_VER_GET_LOCALISED, _fileName, out handle);
-
+            uint infoSize = Interop.Version.GetFileVersionInfoSizeEx(Interop.Version.FileVersionInfoType.FILE_VER_GET_LOCALISED, _fileName, out _);
             if (infoSize != 0)
             {
-                byte[] mem = new byte[infoSize];
-                fixed (byte* memPtr = &mem[0])
+                void* memPtr = NativeMemory.Alloc(infoSize);
+                try
                 {
-                    IntPtr memIntPtr = new IntPtr((void*)memPtr);
+                    IntPtr memIntPtr = new IntPtr(memPtr);
```

Comment:
You can also change the PInvoke signature to have `void*` instead of IntPtr while you are on it.